### PR TITLE
#25 integration tests

### LINKED ISSUE #25: Add Integration Tests target
To set up end-to-end testing passing through the server, we could use a separate test target for integration testing.
- The target should be configured NOT to run together with the unit tests.
- It can be used to test real world scenarios, happy paths and corner cases.
- It should be possible to inject the server URL, if we want to switch to run on a local server or staging server.

### PR DESCRIPTION
**Pull request by MisterVants**
closes #25

Adds an integration test target with a Relay test for subscribe APIs.

The end-to-end payload test using the staging server is still flaky, sometimes it passes, sometimes it fails, still need to understand why.

### CODE REVIEW

File: Sources/WalletConnect/Relay/Relay.swift [hunk 1]
```diff
@@ -30,15 +33,27 @@ class Relay: Relaying {
...
         self.crypto = crypto
-        setUpTransport()
+        setUpBindings()
     }
 
+    private func setUpBindings() {
+        transport.onMessage = { [weak self] payload in
+            self?.handlePayloadMessage(payload)
+        }
+        payloadCancellable = clientSynchJsonRpcPublisher
```
Comment (llbartekll) [inline Sources/WalletConnect/Relay/Relay.swift:51]: @MisterVants  my intention was that `clientSynchJsonRpcPublisher` allows to have many subscribers on clientsynch payloads. And any subscriber is able to filtered out incomming messages by related subscription id - one relay, many subscribers. with `onPayload` approach we can have only one to one relation.

**Review by llbartekll — CHANGES_REQUESTED** [review]
I think we need to synch with those changes.

File: Sources/WalletConnect/Relay/Relay.swift (hunk 1 above)
Comment (llbartekll) [reply Sources/WalletConnect/Relay/Relay.swift:51]: check [this](https://github.com/WalletConnect-Labs/WalletConnectSwiftV2/blob/a3089ae5d49b374734d3ca1fe26c8b1b724c6eeb/Sources/WalletConnect/Subscription/WCSubscribing.swift#L62) out

File: Tests/IntegrationTests/RelayTests.swift [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        let relay = makeRelay()
+        let expectation = expectation(description: "subscribe call must succeed")
+        
+        _ = try? relay.subscribe(topic: String.randomTopic()) { result in
+            result.isSuccess ? expectation.fulfill() : XCTFail("subscribe result must be a success")
+        }
+        waitForExpectations(timeout: defaultTimeout, handler: nil)
+    }
+    
+    func testEndToEndPayload() {
```
[the remark this answers is not in the document]
Comment (MisterVants) [reply Tests/IntegrationTests/RelayTests.swift:23]: It's not related to timeout, I tried bigger timeouts and it keeps failing sometimes, I couldn't find the answer yet. It just that the second subscription payload is never received and it stops there.

[the remark this answers is not in the document]
Comment (llbartekll) [reply Tests/IntegrationTests/RelayTests.swift:23]: hmm interesting 🤔  maybe we should debug it with proxy...?

File: Sources/WalletConnect/Relay/Relay.swift (hunk 1 above)
Comment (MisterVants) [reply Sources/WalletConnect/Relay/Relay.swift:51]: You're right, I overlooked that for the test, the relay needs to notify multiple subscribers. I'll revert that.

File: Sources/WalletConnect/Relay/Relay.swift [hunk 3]
```diff
@@ -116,76 +131,45 @@ class Relay: Relaying {
...
-    }
-
-    private func onPayload(_ payload: String) {
-        if let request = getClientSubscriptionRequest(from: payload) {
+    private func handlePayloadMessage(_ payload: String) {
+        if let request = tryDecode(JSONRPCRequest<RelayJSONRPC.SubscriptionParams>.self, from: payload),
+           request.method == RelayJSONRPC.Method.subscription.rawValue {
             manageSubscriptionRequest(request)
-        } else if let response = getRequestAcknowledgement(from: payload) {
+        } else if let response = tryDecode(JSONRPCResponse<Bool>.self, from: payload) {
```
[the remark this answers is not in the document]
Comment (MisterVants) [reply Sources/WalletConnect/Relay/Relay.swift:138]: You got a point, I think multiple functions doing the same thing is not good, but there is this semantic issue. Do you have any suggestion?

[the remark this answers is not in the document]
Comment (llbartekll) [reply Sources/WalletConnect/Relay/Relay.swift:138]: personally I would leave named function with a single line tryDecode body. like:
getRequestAcknowledgement() {tryDecode(JSONRPCResponse<Bool>.self, from: payload)}
but it is a small thing, won't argue :)

File: Sources/WalletConnect/Relay/Relay.swift (hunk 1 above)
Comment (llbartekll) [reply Sources/WalletConnect/Relay/Relay.swift:51]: 👍

File: Sources/WalletConnect/Relay/Relay.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (MisterVants) [reply Sources/WalletConnect/Relay/Relay.swift:138]: We can also use a local `typealias` to specify the types, like `typealias RequestAcknowledgement = JSONRPCResponse<Bool>` for example, what do you think?

[the remark this answers is not in the document]
Comment (llbartekll) [reply Sources/WalletConnect/Relay/Relay.swift:132]: sounds good to me

**Review by llbartekll — APPROVED** [review]
